src/url.js

```diff
@@ -649,3 +649,36 @@ export function checkCorsUrl(url) {
 export function tryDecodeUriComponent(component, opt_fallback) {
   return tryDecodeUriComponent_(component, opt_fallback);
 }
+
+/**
+ * Extract and return the domain from the given url string via regex
+ * Ex.  'http://google.com?test' -> 'http://google.com'
+ *      'https://google.com/test1/test2' -> 'https://google.com'
+ *      'http://localhost:8000/test' -> 'http://localhost:8000'
+ * @param {string} url
+ * @return {string|undefined}
+ */
+export function extractDomain(url) {
```

Comment:
Could you please help running a test and see if we support relative url from analytics requests. My guess is that we do. Thank you!

Reply:
@torch2424 `parseUrlWithA` uses URL constructor/creating an `a` element. Using regex seems to be a lot faster (https://jsperf.com/hostname-from-url/36) if we just need the domain.

Reply:
+1 to using browser URL parsing. 

> Parsing and canonicalization of URLs is extremely error-prone, and mistakes can introduce both spoofing vulnerabilities and lower-level problems that can lead to vulnerabilities as severe as remote code execution.

https://chromium.googlesource.com/chromium/src/+/master/docs/security/url_display_guidelines/url_display_guidelines.md#Use-Only-Security_Reviewed-Libraries-to-Canonicalize-and-Parse-Strings-into-URLs

Reply:
Got it! Will use browser URL parsing instead